gio/gappinfo.c

```diff
@@ -745,6 +751,38 @@ open_uri_done (GObject      *source,
   g_variant_unref (res);
 }
 
+static char *
+real_uri_for_portal (const char          *uri,
+                     GAppLaunchContext   *context,
+                     GCancellable        *cancellable,
+                     GAsyncReadyCallback  callback,
+                     gpointer             user_data,
+                     GError             **error)
+{
+  GFile *file = NULL;
+  char *real_uri = NULL;
+
+  file = g_file_new_for_uri (uri);
+  if (g_file_is_native (file))
+    {
+      real_uri = g_document_portal_add_document (file, error);
+      g_object_unref (file);
```

Comment:
The "smart" pointers are provided by GLib and not used in glib itself other than for tests. You can check that with a simple `git grep` :)